components/ontology-access/api/src/main/java/org/phenotips/ontology/internal/solr/AbstractHGNCSolrOntologyService.java

```diff
@@ -0,0 +1,339 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.phenotips.ontology.internal.solr;
+
+import org.phenotips.obo2solr.ParameterPreparer;
+import org.phenotips.obo2solr.TermData;
+import org.phenotips.ontology.OntologyTerm;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.Consts;
+import org.apache.http.HttpHeaders;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.entity.ContentType;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.common.SolrDocument;
+import org.apache.solr.common.SolrDocumentList;
+import org.apache.solr.common.SolrException;
+import org.apache.solr.common.SolrInputDocument;
+
+import net.sf.json.JSONException;
+import net.sf.json.JSONObject;
+import net.sf.json.JSONSerializer;
+
+/**
+ * Ontologies processed from HGNC file share much of the processing code.
+ *
+ * @since 1.2RC1
+ * @version $Id$
+ */
+public abstract class AbstractHGNCSolrOntologyService extends AbstractSolrOntologyService
+{
+    /**
+     * The name of the Alternative ID field, used for older aliases of updated HPO terms.
+     */
+    protected static final String ALTERNATIVE_ID_FIELD_NAME = "alt_id";
+
+    protected static final String VERSION_FIELD_NAME = "version";
+
+    protected static final String SIZE_FIELD_NAME = "size";
+    protected static final String ROWS_FIELD_NAME = "rows";
+
+    private static final String FIELD_VALUE_SEPARATOR = "\\t";
+
+    private String infoServiceURL = "http://rest.genenames.org/info";
+
+    /** Performs HTTP requests to the remote REST service. */
+    private final CloseableHttpClient client = HttpClients.createSystem();
+
+    /** The number of documents to be added and committed to Solr at a time. */
+    protected abstract int getSolrDocsPerBatch();
+
+    private Map<String, TermData> transform(String ontologyUrl, Map<String, Double> fieldSelection)
+    {
+        URL url;
+        try {
+            url = new URL(ontologyUrl);
+        } catch (MalformedURLException ex) {
+            return null;
+        }
+        return transform(url, fieldSelection);
+    }
+
+    private Map<String, TermData> transform(URL url, Map<String, Double> fieldSelection)
+    {
```

Comment:
Have you considered using https://commons.apache.org/proper/commons-csv/ for parsing the input?